src/app/mysterion.js

```
@@ -59,60 +75,71 @@ class Mysterion {
     this.communication = new MainCommunication(this.messageBus)
+    this.communication.onCurrentIdentityChange((identity) => {
+      this.bugReporter.setUser(identity)
+    })
 
     try {
       await onFirstEvent(this.communication.onRendererBooted.bind(this.communication))
     } catch (e) {
-      console.error(e)
-      this.bugReporter.captureException(e)
       // TODO: add an error wrapper method
-      throw new Error('Failed to load app.')
+      throw new Error('Failed to load app. ' + e)
     }
 
     // make sure terms are up to date and accepted
     // declining terms will quit the app
     if (!termsAccepted) {
+      logInfo('Accepting terms')
       try {
         const accepted = await this.acceptTerms()
         if (!accepted) {
-          console.log('Terms were refused. Quitting.')
+          logInfo('Terms were refused. Quitting.')
           app.quit()
           return
         }
       } catch (e) {
-        this.bugReporter.captureException(e)
-        return this.communication.sendRendererShowErrorMessage(e.message)
+        this.communication.sendRendererShowErrorMessage(e.message)
+        throw new Error('Failed to accept terms. ' + e)
       }
     }
 
+    logInfo('Redirecting to application window')
+    this.window.resize(this.config.windows.app)
+
     // checks if daemon is installed or daemon file is expired
     // if the installation fails, it sends a message to the renderer window
     if (this.installer.needsInstallation()) {
+      logInfo("Installing 'mysterium_client' process")
       try {
         await this.installer.install()
       } catch (e) {
-        this.bugReporter.captureException(e)
-        console.error(e)
-        return this.communication.sendRendererShowErrorMessage(translations.daemonInstallationError)
+        this.communication.sendRendererShowErrorMessage(translations.daemonInstallationError)
+        throw new Error("Failed to install 'mysterium_client' process. " + e)
```

We should not create a new instance of `Error` here - stacktrace of original error is lost here, which provides details of where error happened.

---

There is TODO for that, not in PR scope

---

Which TODO are you referring? I'm not talking about some unrelated existing issues - this change brakes stacktrace.

---

It's old convention in this file with TODO registered. Fix fill come in upcomming separate PR